deploy/test-environments/fleet_api/src/api/common_api.py

```diff
@@ -142,3 +142,111 @@ def get_build_info(version: str, is_snapshot: bool) -> str:
             f"API call failed, status code {api_ex.status_code}. Response: {api_ex.response_text}",
         )
         return ""
+
+
+def get_stack_latest_version():
+    """
+    Retrieve the latest version of the stack from the Elastic snapshots API.
+
+    Returns:
+        str: The latest version of the stack.
+
+    Raises:
+        APICallException: If the API call to retrieve the version fails.
+
+    """
+    url = "https://snapshots.elastic.co/latest/master.json"
+    try:
+        response = perform_api_call(
+            method="GET",
+            url=url,
+        )
+        response_obj = munchify(response)
+        return response_obj.version
+
+    except APICallException as api_ex:
+        logger.error(
+            f"API call failed, status code {api_ex.status_code}. Response: {api_ex.response_text}",
+        )
+        return ""
+
+
+def get_cloud_security_posture_version(cfg: Munch, prerelease: bool = True) -> str:
+    """
+    Retrieve the version of the cloud_security_posture package.
+
+    Args:
+        cfg (Munch): Configuration object containing Kibana URL, authentication details, etc.
+        prerelease (bool, optional): Flag indicating whether to include
+                                        prerelease versions.Defaults to True.
+
+    Returns:
+        str: The version of the cloud_security_posture package, or None if the API call fails.
+
+    """
+    url = f"{cfg.kibana_url}/api/fleet/epm/packages"
+
+    request_params = {
+        "prerelease": prerelease,
+    }
+
+    try:
+        response = perform_api_call(
+            method="GET",
+            url=url,
+            auth=cfg.auth,
+            params={"params": request_params},
+        )
+        response_obj = munchify(response)
+        cloud_security_posture_version = None
+        for package in response_obj.response:
+            if package["name"] == "cloud_security_posture":
```

Comment:
nit: to avoid missing value errors
```suggestion
            if package.get("name", "")== "cloud_security_posture":
```